examples/move/reviews_rating/sources/service.move

```diff
@@ -0,0 +1,301 @@
+        };
+    }
+
+    /// Finds the index of a review in top_reviews
+    fun find_idx(service: &Service, total_score: u64): u64 {
+        let len = service.top_reviews.length();
+        let mut i = 0;
+        let mut idx = 0;
+        while (i < len) {
+            idx = len - i - 1;
+            let review_id = *vector::borrow(&service.top_reviews, idx);
+            if (get_total_score(service, review_id) > total_score) {
+                return idx + 1
+            };
+            i = i + 1;
+        };
+        idx
+    }
+
+    /// Gets the total score of a review
+    fun get_total_score(service: &Service, review_id: ID): u64 {
+        let review = object_table::borrow(&service.reviews, review_id);
+        review::get_total_score(review)
+    }
+
+    /// Distributes rewards
+    public fun distribute_reward(
+        cap: &AdminCap,
+        service: &mut Service,
+        ctx: &mut TxContext
+    ) {
+        assert!(cap.service_id == object::uid_to_inner(&service.id), EInvalidPermission);
+        // distribute a fixed amount to top MAX_REVIEWERS_TO_REWARD reviewers
+        let mut len = vector::length(&service.top_reviews);
+        if (len > MAX_REVIEWERS_TO_REWARD) {
+            len = MAX_REVIEWERS_TO_REWARD;
+        };
+        // check balance
+        assert!(balance::value(&service.reward_pool) >= (service.reward * len), ENotEnoughBalance);
+        let mut i = 0;
+        while (i < len) {
+            let sub_balance = balance::split(&mut service.reward_pool, service.reward);
+            let reward = coin::from_balance(sub_balance, ctx);
+            let review_id = vector::borrow(&service.top_reviews, i);
+            let record = df::borrow<ID, ReviewRecord>(&service.id, *review_id);
+            transfer::public_transfer(reward, record.owner);
+            i = i + 1;
+        };
+    }
+
+    /// Adds coins to reward pool
+    public fun top_up_reward(
+        service: &mut Service,
+        coin: Coin<SUI>
+    ) {
+        balance::join(&mut service.reward_pool, coin::into_balance(coin));
+    }
+
+    /// Mints a proof of experience for a customer
+    public fun generate_proof_of_experience(
+        cap: &AdminCap,
+        service: &Service,
+        recipient: address,
+        ctx: &mut TxContext
+    ) {
+        // generate an NFT and transfer it to customer who can use it to write a review with higher score
+        assert!(cap.service_id == object::uid_to_inner(&service.id), EInvalidPermission);
+        let poe = ProofOfExperience {
+            id: object::new(ctx),
+            service_id: cap.service_id
+        };
+        transfer::transfer(poe, recipient);
+    }
+
+    /// Removes a review (only moderators can do this)
+    public fun remove_review(
+        _: &Moderator,
+        service: &mut Service,
+        review_id: ID,
+    ) {
+        assert!(service.reviews.contains(review_id), ENotExists);
+        let record = df::remove<ID, ReviewRecord>(&mut service.id, review_id);
```

Comment:
```suggestion
        let record = df::remove(&mut service.id, review_id);
```

Reply:
For this one, the compiler could not infer type so I had to keep the annotation